C7/UIElements/Popups/BuildCityDialog.cs

```diff
@@ -8,7 +8,7 @@ public class BuildCityDialog : Popup
 	public BuildCityDialog()
     {
 		alignment = BoxContainer.AlignMode.End;
-		margins = new Margins();
+		margins = new Margins(right: 10); // 10px margin on the right
```

Comment:
Trying things out manually, I noticed the build city dialog wasn't 10 pixels in from the right.  Playing around with it, I figured out that it's actually required to be right: -10 to be 10 pixels left of the right edge.

This might be related, but how are the margins actually applied?  I see them declared on Popup.cs, and changing them has an outcome, so they must be... inherited from Godot somehow?  But that's a bit confusing since Margins is a class defined _within_ Popup.cs, so how could Godot be inheriting that behavior?

Not sure if that confusion also contains a hint as to why right must be -10 instead of +10, but the confusion is why I can't tell you why it needs to be -10 (and wouldn't have guessed it if I hadn't tried 50 instead of 10 and noticed it went the unexpected way).

Reply:
They're taken from the Margins class and applied to the Popupoveraly manually in the popup overlay code before adding the actual Popup as a child. This might not be necessary though if we can set margins on the Popup despite it not being the top level component. As for the negative margins I'm not sure since whatever we set in margins is set directly, so I'll look into that some more

Reply:
@pcen - What are your thoughts on setting the margin to -10, and opening a separate issue to investigate the why and potentially making it more intuitive?  I'd like to build upon your changes in this PR with another popup, and this doesn't seem like a big enough issue to justify keeping the entire popup refactor out of the Development branch.

Reply:
Makes sense, I'll make those changes and we can get this merged